enterprise_access/apps/bffs/tests/test_handlers.py

```diff
@@ -0,0 +1,125 @@
+from django.test import RequestFactory, TestCase
+from faker import Faker
+from rest_framework.exceptions import ValidationError
+
+from enterprise_access.apps.bffs.context import HandlerContext
+from enterprise_access.apps.bffs.handlers import BaseHandler, BaseLearnerPortalHandler, DashboardHandler
+from enterprise_access.apps.core.tests.factories import UserFactory
+
+
+class TestBaseHandler(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.user = self.mock_user
+        self.context = HandlerContext(self.request)
+        self.base_handler = BaseHandler(self.context)
+
+    def test_base_handler_uninitialized_load_and_process(self):
+        base_handler = self.base_handler
+        with self.assertRaises(NotImplementedError):
+            base_handler.load_and_process()
+
+    def test_base_handler_add_error_serializer(self):
```

Comment:
nit: might remove the `_serializer` suffix as the handler's `add_error` is largely just a wrapper around the `HandlerContext`'s `add_error`. Given we already have sufficient coverage on the `HandlerContext`'s `add_error` and `add_warning` methods, we probably don't necessarily need to re-test the passing of extra kwargs like `status`.

Similar feedback to testing `is_valid`. Given the serializer `is_valid` stuff is already tested via `test_context.py`, I'm not sure we need to retest its logic here. The most important thing to test here, IMHO, is just that the `BaseHandler` is wired up to the `HandlerContext`'s `add_error` and `add_warning` from a happy path perspective (valid kwargs).